presto-main/src/main/java/io/prestosql/operator/CommonTableExecutionContext.java

```diff
@@ -264,4 +277,40 @@ public CTEDoneException()
             super();
         }
     }
+
+    @Override
+    public Object capture(BlockEncodingSerdeProvider serdeProvider) {
+        BlockEncodingSerde blockSerde = serdeProvider.getBlockEncodingSerde();
+        CommonTableExecutionContextState myState = new CommonTableExecutionContextState();
+
+        myState.queueCnt = queueCnt;
+        myState.maxPrefetchQueueSize = maxPrefetchQueueSize;
+        myState.maxMainQueueSize = maxMainQueueSize;
+        myState.taskCount = taskCount;
+        return myState;
+    }
+
+    @Override
+    public void restore(Object state, BlockEncodingSerdeProvider serdeProvider)
```

Comment:
I don't know how a recovery can occur without Feeders List and PrefetchedQueue?
this seems like incomplete PR or WIP; it should be indicated so in the PR TItle.